packages/desktop-client/src/components/accounts/MobileAccountDetails.js

```diff
@@ -16,62 +16,61 @@ import SearchAlternate from 'loot-design/src/svg/v2/SearchAlternate';
 
 import { TransactionList } from './MobileTransaction';
 
-class TransactionSearchInput extends React.Component {
-  state = { text: '' };
+function TransactionSearchInput({ accountName, onSearch }) {
+  const [text, setText] = useState('');
 
-  performSearch = () => {
-    this.props.onSearch(this.state.text);
-  };
+  const performSearch = useCallback(() => {
+    onSearch(text);
+  }, [text, onSearch]);
 
-  onChange = text => {
-    this.setState({ text }, this.performSearch);
-  };
-
-  render() {
-    const { accountName } = this.props;
-    const { text } = this.state;
+  const onChange = useCallback(
+    text => {
+      setText(text);
+      performSearch();
+    },
+    [performSearch],
+  );
```

Comment:
Does this still work?

Before:
`setText` and after the state changed -> `performSearch`

After:
`setText` and in parallel (same CD cycle) `performSearch` with old `text` value